service/src/main/java/bio/terra/workspace/db/DbRetryUtils.java

```diff
@@ -0,0 +1,80 @@
+package bio.terra.workspace.db;
+
+import bio.terra.common.db.DatabaseRetryUtils;
+import java.time.Duration;
+import java.util.List;
+import org.springframework.dao.RecoverableDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+/**
+ * This class contains wrappers around JDBC methods which run them with retries. Only DB exceptions
+ * considered retryable ({@link RecoverableDataAccessException} and {@link
+ * org.springframework.dao.TransientDataAccessException}) are retried. See Spring documentation of
+ * these exceptions for more information:
+ * https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/dao/DataAccessException.html
+ *
+ * <p>Retries are with no delay, up to a certain maximum number of retries.
+ */
+public class DbRetryUtils {
+
+  public static int MAX_RETRIES = 10;
+
+  public static int update(
+      NamedParameterJdbcTemplate jdbcTemplate, String sql, MapSqlParameterSource params) {
+    try {
+      return DatabaseRetryUtils.executeAndRetry(
+          () -> jdbcTemplate.update(sql, params), Duration.ZERO, MAX_RETRIES);
```

Comment:
I zero duration is too short. I'd allow at least ~250ms to allow the conflicting transaction to complete. Besides, I don't think any of our endpoints are so performance sensitive that we need to be concerned at that amount of time.

What has been your experience in testing this?

Reply:
sgtm. I hadn't seen any failures with the instant retries, but that seems like a risk for flakiness if we have longer running queries in the future.